sql/src/main/java/io/crate/execution/engine/window/AggregateToWindowFunctionAdapter.java

```diff
@@ -110,10 +115,14 @@ private void removeSeenRowsFromAccumulatedState(WindowFrameState frame,
             for (int j = 0, expressionsSize = expressions.size(); j < expressionsSize; j++) {
                 expressions.get(j).setNextRow(row);
             }
-            accumulatedState = aggregationFunction.removeFromAggregatedState(ramAccountingContext,
-                                                                             accumulatedState,
-                                                                             args);
-
+            //noinspection unchecked
+            if (InputCondition.matches(filter)) {
```

Comment:
could we maybe cast to `Input<Boolean>` early/here already https://github.com/crate/crate/blob/master/sql/src/main/java/io/crate/expression/InputFactory.java#L205 and adjust the `AggregationContext.filter` type? this would avoid multiple casts (also at the AggregateCollector, etc)